ldclient/client.py

```diff
@@ -249,14 +251,19 @@ def __start_up(self, start_wait: float):
         self.__hooks_lock = ReadWriteLock()
         self.__hooks = self._config.hooks + plugin_hooks  # type: List[Hook]
 
-        # Initialize data system (FDv1) to encapsulate v1 data plumbing
-        from ldclient.impl.datasystem.fdv1 import (  # local import to avoid circular dependency
-            FDv1
-        )
+        datasystem_config = self._config.datasystem_config
+        if datasystem_config is None:
+            # Initialize data system (FDv1) to encapsulate v1 data plumbing
+            from ldclient.impl.datasystem.fdv1 import (  # local import to avoid circular dependency
+                FDv1
+            )
+
+            self._data_system: DataSystem = FDv1(self._config)
+        else:
+            self._data_system = FDv2(datasystem_config, disabled=self._config.offline)
 
-        self._data_system = FDv1(self._config)
         # Provide flag evaluation function for value-change tracking
-        self._data_system.set_flag_value_eval_fn(
+        self._data_system.set_flag_value_eval_fn(  # type: ignore
```

Comment:
These ignores are because the data system protocol doesn't expose these methods, but both fdv1 and fdv2 implementations do. So we have to tell the linter to trust us in a few places.